File: python/sample.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

"""
Copyright: (C) 2017, Nextpertise B.V.
Author:    Leo Noordergraaf
"""

import NextpertiseAPI as api

USERNAME = 'your username'
PASSWORD = 'your password'

ZIPCODE = 'your zipcode'
HOUSENR = your house number

class MyZipcode(object):
    ZIPCODE_TRANSLATE = {
        'KPNWBA' : 'KPN (WBA)',
        'KPNWEAS': 'KPN (Weas)',
        'EUROFIBER': 'Eurofiber',
        'TELE2': 'Tele2',
        'ADSL' : 'Zakelijk ADSL',
        'VDSL': 'Zakelijk VDSL',
        'SDSL': 'Zakelijk SDSL',
        'SDSL.bis': 'Zakelijk SDSL.bis',
        'Fiber': 'Zakelijk Glasvezel'
    }

    def __init__(self):
        auth = api.AuthPLAIN(USERNAME, PASSWORD)
        self.client = api.BasicBroadband(auth)

    def zipcode(self, zipcode, housenr, housenrext=None):
        # get zipcode data
        data = self.client.zipcode(zipcode, housenr, housenrext)
        
        if 'available' in data:
            # scan all items
            # replace integer speed in kb/s with string and units.
            for k,v in data['available'].items():
                for kk,vv in v.items():
                    if 'max_download' in vv:
                        vv['max_download'] = self._Kb2Mb(vv['max_download'])
                    if 'max_upload' in vv:
                        vv['max_upload'] = self._Kb2Mb(vv['max_upload'])
            
            # scan all items
            # replace text labels present as key in ZIPCODE_TRANSLATE
            # with their values.
            for k,v in data['available'].items():
                for kk,vv in v.items():
                    tk = self._translate(kk)
                    if kk != tk:
                        v[tk] = v[kk]
                        del v[kk]
                tk = self._translate(k)
                if k != tk:
                    data['available'][tk] = data['available'][k]
                    del data['available'][k]
        return data

    def _translate(self, txt):
        if txt in self.ZIPCODE_TRANSLATE:
            return self.ZIPCODE_TRANSLATE[txt]
        return txt

    def _Kb2Mb(self, val):
        val = int(val)
        if val < 1024:
            return '%d Kb' % val
        return '%d Mb' % round(val / 1024)

if __name__ == '__main__':
    client = MyZipcode()
    print(client.zipcode(ZIPCODE, HOUSENR))

File: php/sample.php
<?php

require __DIR__ . '/vendor/autoload.php';
require __DIR__ . '/NextpertiseAPI/init.php';

class MyZipcode {
    
    private $client = null;
    private $provider_mapper = array(
        'KPNWBA' => 'KPN (wba)',
        'KPNWEAS' => 'KPN (weas)',
        'EUROFIBER' => 'EuroFiber',
        'TELE2' => 'Tele2',
    );
    private $technology_mapper = array(
        'ADSL' => 'Zakelijke ADSL',
        'VDSL' => 'Zakelijke VDSL',
        'SDSL' => 'Zakelijke SDSL',
        'SDSL.bis' => 'Zakelijke SDSL.bis',
        'Fiber' => 'Zakelijke Glasvezel',
    );
    
    function __construct($username, $password, $testing = false) {
        $this->client = new NextpertiseAPI\BasicBroadband($username, $password, $testing);
    }
    
    function zipcode($zipcode, $housenr, $housenrext = '') {
        $ret = $this->client->zipcode($zipcode, $housenr, $housenrext);
        foreach($ret['available'] as $provider => $technologies) {
            foreach($ret['available'][$provider] as $technology => $attributes) {
                // Add max_upload_str/max_download_str in readable string format
                $ret['available'][$provider][$technology]['max_upload_str'] = $this->_Kb2Str($ret['available'][$provider][$technology]['max_upload']);
                $ret['available'][$provider][$technology]['max_download_str'] = $this->_Kb2Str($ret['available'][$provider][$technology]['max_download']);
                // Update technology type with mapped technology type
                if(isset($this->technology_mapper[$technology])) {
                    $ret['available'][$provider] = $this->_change_key($ret['available'][$provider], $technology, $this->technology_mapper[$technology]);   
                }
            }
            // Update provider with mapped provider
            if(isset($this->provider_mapper[$provider])) {            
                $ret['available'] = $this->_change_key($ret['available'], $provider, $this->provider_mapper[$provider]);
            }
        }
        return $ret;   
    }
    
    function _change_key($array, $old_key, $new_key) {
        if( ! array_key_exists( $old_key, $array ) )
            return $array;
    
        $keys = array_keys($array);
        $keys[array_search($old_key, $keys)] = $new_key;
    
        return array_combine($keys, $array);
    }
    
    function _Kb2Str($val) {
        if($val < 1024)
            return sprintf('%d Kb', $val);
        if($val < 1000000)
            return sprintf('%d Mb', round($val / 1024) );
        return sprintf('%d Gb', round($val / 1024 / 1024) );
    }
}

$zipcodeHandler = new MyZipcode('USERNAME HERE', 'PASSWORD HERE');
$result = $zipcodeHandler->zipcode('1000AA', 1);
print_r($result);